### PR TITLE
fix(tooltip): use core positioning utils

### LINKED ISSUE #188: [Tooltip] multiple Tooltips not working
When using multiple Tooltips not working `[sideOffset]` and others inputs on `rdxTooltipContent`

Example:
https://sb-primitives.radix-ng.com/?path=/story/primitives-tooltip--multiple

### PR DESCRIPTION
**Pull request by pawel-twardziak**
### Description

Closes #188 

this does three things:

1. refactor(popover): move positioning stuff to core lib
2. refactor(popover): make arrow pos depends on itself dimensions change
3. fix(tooltip): use core positioning utils - **this one fixes the issue**

### CODE REVIEW

**Comment by pawel-twardziak** [comment]
@mrfrac I slightly adjusted the code to the `popover` code. Check if it's ok for you.
DW @pimenovoleg

File: packages/primitives/core/src/positioning/utils.ts [hunk 1]
```diff
@@ -1,25 +1,52 @@
...
+            if (alignOffset) {
+                position.offsetY = alignOffset;
+            }
+        }
+    }
+    return position;
+}
 
 let allPossibleConnectedPositions: RdxAllPossibleConnectedPositions;
 export function getAllPossibleConnectedPositions() {
```
Comment (pimenovoleg) [inline packages/primitives/core/src/positioning/utils.ts:40]:
```
let allPossibleConnectedPositions: RdxAllPossibleConnectedPositions | null = null;

getAllPossibleConnectedPositions(): {
    if (allPossibleConnectedPositions === null) {
        allPossibleConnectedPositions = new Map();

        for (const [side, aligns] of Object.entries(RDX_POSITIONS)) {
            if (!aligns) continue;
            for (const [align, position] of Object.entries(aligns)) {
                allPossibleConnectedPositions.set(`${side}|${align}`, position);
            }
        }
    }

    return allPossibleConnectedPositions;
}
```

The **current** implementation used `Object.keys`, which required additional lookups for each key to access its corresponding value.

The **refactored** implementation leverages `Object.entries`, which provides both key and value directly in a single iteration:
```typescript
for (const [side, aligns] of Object.entries(RDX_POSITIONS)) {
    for (const [align, position] of Object.entries(aligns)) {
        allPossibleConnectedPositions.set(`${side}|${align}`, position);
    }
}
```

**Impact**
- **Complexity**: No changes in the algorithmic complexity (still O(n)O(n)).
- improved **Performance**: fewer object property lookups reduce execution time, especially with large datasets.
- direct access to keys and values makes the code easier to understand and maintain.

**Comment by pimenovoleg** [comment]
> @mrfrac I slightly adjusted the code to the `popover` code. Check if it's ok for you. DW @pimenovoleg I've moved the positioning stuff from the `popover` code to the `core` code. Ok?

Thanks for pointing this out. Moving the positioning logic to core sounds good to me—makes it reusable for other components as well.

**Comment by mrfrac** [comment]
> @mrfrac I slightly adjusted the code to the `popover` code. Check if it's ok for you. DW @pimenovoleg I've moved the positioning stuff from the `popover` code to the `core` code. Ok?

Hi! I'll definitely watch it today! Thanks :)

File: packages/primitives/core/src/positioning/types.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { ConnectionPositionPair } from '@angular/cdk/overlay';
+
+export enum RdxSide {
```
Comment (mrfrac) [inline packages/primitives/core/src/positioning/types.ts:3]: Naming like `RdxPositionSide` / `RdxPositioningSide` will provide more context for later use, imo

**Review by mrfrac — APPROVED** [review]
@pawel-twardziak tooltip is ok! thanks

File: packages/primitives/core/src/positioning/utils.ts (hunk 1 above)
Comment (pawel-twardziak) [reply packages/primitives/core/src/positioning/utils.ts:40]: Great catch! Thanks @pimenovoleg 🚀

File: packages/primitives/core/src/positioning/types.ts (hunk 2 above)
Comment (pawel-twardziak) [reply packages/primitives/core/src/positioning/types.ts:3]: Nice, will change it 💯

**Comment by pawel-twardziak** [comment]
@pimenovoleg @mrfrac I've also replaced all the `window` and `inject(DOCUMENT)` with the appropirate `injectWindow()` and `injectDocument()` within `popover` and `tooltip`

**Comment by pawel-twardziak** [comment]
@pimenovoleg @mrfrac I've also replaced all the `window` and `inject(DOCUMENT)` with the appropirate `injectWindow()` and `injectDocument()` within `popover` and `tooltip`

**Review by pimenovoleg — APPROVED** [review]

**Pull request merged**